public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/DimensionFields.test.tsx

```diff
@@ -18,342 +19,365 @@ const variableOptionGroup = {
 };
 const user = userEvent.setup();
 
-describe('Azure Monitor QueryEditor', () => {
-  const mockDatasource = createMockDatasource();
+const tests = [
```

Comment:
Totally agree that the array approach doesn't feel as comfortable here as it does in Go. However, I think it's worth the tradeoff here because it lets us verify _every_ test with the feature flag on and the feature flag off without changing the existing tests _too_ much. Also, if we eventually don't need the feature flag I'd be all for removing the array pattern here as part of the feature flag removal. TLDR;  the array pattern is very much tied to the feature flag.